flax2bnn.py

```diff
@@ -0,0 +1,157 @@
+
+    def map_distribution(self, config: dict) -> dist.Distribution:
+        """Map the distribution from string to callable."""
+        if config['dist'] == 'Normal':
+            return dist.Normal(scale=config['sd'])
+        elif config['dist'] == 'Laplace':
+            return dist.Laplace(scale=config['sd'])
+        else:
+            raise ValueError(f'Distribution {config["dist"]} not implemented')
+
+    def set_constant_prior(
+        self, params: dict, distr: dist.Distribution = dist.Normal()
+    ) -> dict:
+        """
+        Set a constant prior for the parameters of the model.
+
+        Args:
+        params: dict
+        distr: dist.Distribution
+
+        Returns:
+        dict: dictionary of priors
+        """
+        return {k: distr for k in get_flattened_keys(params)}
+
+    def set_prior(self):
+        """
+        Set the prior for the parameters of the model.
+
+        Returns:
+        dict: dictionary of priors with the keys being the concatenated parameter names.
+        """
+        if self.prior_config['scheme'] == 'equal':
+            distr = self.map_distribution(self.prior_config['details'])
+            return self.set_constant_prior(self.params, distr=distr)
+        else:
+            raise ValueError(f'Scheme {self.prior_config["scheme"]} not implemented')
+
+    def log_prior(self, params: dict):
+        """Log prior for the parameters of the model."""
+        prior_logprobs = jnp.array([0.0])
+        for k in self.priorsdict.keys():
+            levels = k.split('.')
+            prior_logprobs = jnp.concatenate(
+                [
+                    prior_logprobs,
+                    self.priorsdict[k].log_prob(jnp.ravel(get_by_path(params, levels))),
+                ]
+            )
+        return jnp.sum(prior_logprobs)
+
+    def log_likelihood(
+        self, params: dict, X: jnp.ndarray, Y: jnp.ndarray, type: str = 'regr'
+    ):
+        """
+        Evaluate Log likelihood of the model.
+
+        Args:
+        params: dict - parameters of the model (potentially many nested levels)
+        X: jnp.ndarray - input data either full batch or mini-batch of dimensions
+          with the first dimension being the batch size
+        Y: jnp.ndarray - target data either full batch or mini-batch. The dimension has
+            to be the same as the first dimension of X (batch size,). For classification
+            problems, Y is expected to be integer-encoded.
+        type: str - type of learning target (either 'regr' or 'class')
+        """
+        chex.assert_shape(Y, (X.shape[0],))
+        lvals = self.module.apply({'params': params}, X)
+        if type == 'regr':
+            return jnp.sum(
+                dist.Normal(
+                    loc=lvals[..., 0],
+                    scale=jnp.exp(lvals[..., 1]).clip(min=1e-6, max=1e6),
+                ).log_prob(Y)
+            )
+        else:
+            return jnp.sum(dist.Categorical(logits=lvals).log_prob(Y))
```

Comment:
No need for an `else` here, since the function is returned in the `if` statement.
```suggestion
        return jnp.sum(dist.Categorical(logits=lvals).log_prob(Y))
```